From a227f70d964de96121583c733af04ffe69085bd1 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:15 -0400
Subject: [fs] nfsd: call rpc_destroy_wait_queue() from free_client()

Message-id: <1403560314-32525-74-git-send-email-bfields@redhat.com>
Patchwork-id: 84488
O-Subject: [Patch RHEL7 073/172] NFSd: call rpc_destroy_wait_queue() from free_client()
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Mainly to ensure that we don't leave any hanging timers.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 4cb57e3032d4e4bf5e97780e9907da7282b02b0c
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 230d21c..32b699b 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -1107,6 +1107,7 @@ free_client(struct nfs4_client *clp)
   WARN_ON_ONCE(atomic_read(&ses->se_ref));
   free_session(ses);
  }
+ rpc_destroy_wait_queue(&clp->cl_cb_waitq);
  free_svc_cred(&clp->cl_cred);
  kfree(clp->cl_name.data);
  idr_destroy(&clp->cl_stateids);
-- 
1.7.1